packages/contracts/contracts/chugsplash/L2/ChugSplashDeployer.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: evm
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Library Imports */
+import { Lib_ExecutionManagerWrapper } from "../../optimistic-ethereum/libraries/wrappers/Lib_ExecutionManagerWrapper.sol";
+import { Lib_MerkleTree } from "../../optimistic-ethereum/libraries/utils/Lib_MerkleTree.sol";
+
+/**
+ * @title ChugSplashDeployer
+ */
+contract ChugSplashDeployer {
+    
+    /*********
+     * Enums *
+     *********/
+
+    enum ActionType {
+        SET_CODE,
+        SET_STORAGE
+    }
+
+
+    /***********
+     * Structs *
+     ***********/
+
+    struct ChugSplashAction {
+        ActionType actionType;
+        address target;
+        bytes data;
+    }
+
+    struct ChugSplashActionProof {
+        uint256 actionIndex;
+        bytes32[] siblings;
+    }
+
+
+    /*************
+     * Variables *
+     *************/
+
+    // Address that can approve new transaction bundles.
+    address public owner;
+    bytes32 public currentBundleHash;
+    uint256 public currentBundleSize;
+    uint256 public currentBundleTxsExecuted;
+
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+
+    /**
+     * Marks a function as only callable by the owner.
+     */
+    modifier onlyOwner() {
+        // require(
+        //     msg.sender == owner,
+        //     "ChugSplashDeployer: sender is not owner"
+        // );
+        _;
+    }
+
+
+    /********************
+     * Public Functions *
+     ********************/
+
+    /**
+     * Changes the owner. Only callable by the current owner.
+     * @param _owner New owner address.
+     */
+    function setOwner(
+        address _owner
+    )
+        public
+        onlyOwner
+    {
+        owner = _owner;
+    }
+
+    function hasActiveBundle()
+        public
+        view
+        returns (
+            bool
+        )
+    {
+        return (
+            currentBundleHash != bytes32(0)
+            && currentBundleTxsExecuted < currentBundleSize
+        );
+    }
+
+    function approveTransactionBundle(
+        bytes32 _bundleHash,
+        uint256 _bundleSize
```

Comment:
This should also take in a version string I think which we can emit and thereby track the versions of each deployment. Or alternatively there is a version in the bundle